### PR TITLE
Queue

### PR DESCRIPTION
**Pull request by corinnelhh**

### CODE REVIEW

File: test_queue.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    our_queue = Queue(our_list)
+
+    assert our_queue.size == 3
+
+    assert our_queue.dequeue() == "first"
+
+    assert our_queue.head.data == "third"
+    assert our_queue.tail.data == "second"
+
+    assert our_queue.size == 2
```
Comment (markcharyk) [inline test_queue.py:54]: You're missing a test for the scenario where the queue is empty. Your code is obviously showing that you intend an error to be raised, so you should probably test that the appropriate error is, in fact, raised.
